packages/appstore/src/components/modals/cashier-wallet-modal/cashier-wallet-modal-header.tsx

```diff
@@ -0,0 +1,133 @@
+        if (is_dark) return 'IcAppstoreCloseDark';
+        return 'IcAppstoreCloseLight';
+    }, [is_dark, is_demo]);
+
+    const getStylesByClassName = React.useCallback(
+        (class_name: string) => {
+            return classNames(class_name, {
+                [`${class_name}-demo`]: is_demo,
+            });
+        },
+        [is_demo]
+    );
+
+    const getCurrencyIconSize = React.useCallback(() => {
+        // TODO: add p2p and payment_agent check
+        const is_square_icon =
+            ['btc', 'eth', 'ltc', 'usdt', 'eusdt', 'tusdt', 'ust', 'usdc', 'p2p', 'payment_agent'].includes(
+                currency.toLowerCase()
+            ) || is_demo;
+
+        const sizes = {
+            mobile: {
+                width: is_square_icon ? 64 : 48,
+                height: is_square_icon ? 40 : 48,
+            },
+            desktop: {
+                width: is_square_icon ? 128 : 64,
+                height: is_square_icon ? 80 : 64,
+            },
+        };
+
+        const size = is_mobile ? sizes.mobile : sizes.desktop;
+
+        return size;
+    }, [currency, is_demo, is_mobile]);
+
+    const getCurrencyIconProps = React.useCallback(() => {
+        const icon = getWalletCurrencyIcon(is_demo ? 'demo' : currency, is_dark);
+        const size = getCurrencyIconSize();
+
+        return { icon, ...size };
+    }, [currency, getCurrencyIconSize, is_dark, is_demo]);
+
+    const getBackgroundName = React.useCallback(() => {
+        if (is_demo) {
+            return 'demo';
+        }
+        return currency.toLowerCase();
+    }, [is_demo, currency]);
+
+    return (
+        <GradientBackground {...getCashierWalletModalBackgrounds(getBackgroundName())}>
+            <Watermark
+                image='url(/packages/appstore/src/public/images/wallet-demo-bg.svg)'
+                // image='url(https://i.postimg.cc/jqGY1q6R/Style-white.png)' - works fine
+            />
+            <div
+                className={classNames(header_class_name, {
+                    [`${header_class_name}--hide-title`]: !show_wallet_name,
+                })}
+            >
+                <div className={`${header_class_name}__title-wrapper`}>
+                    {show_wallet_name && (
+                        <div className={classNames(`${header_class_name}__title`)}>
+                            <Text
+                                size={is_mobile ? 'xs' : 's'}
+                                as='span'
+                                className={getStylesByClassName(`${header_class_name}__title-wallet`)}
+                            >
+                                {is_demo ? 'Demo' : ''} {currency} {localize('Wallet')}
+                            </Text>
+                            {is_demo ? (
+                                <Badge type='contained' background_color='blue' label='Demo' />
+                            ) : (
+                                <Badge type='bordered' label='SVG' />
```

Comment:
@george-usynin-deriv This `label`  should be dynamic and get from landing_company_shortcode